lib/modules/dosomething/dosomething_api/resources/member_resource.inc

```diff
@@ -52,13 +52,35 @@ function _member_resource_defintion() {
         'file' => array(
           'type' => 'inc',
           'module' => 'dosomething_api',
-          'name' => 'resources/member.resource',
+          'name' => 'resources/member_resource',
         ),
         'callback' => '_member_resource_get_member_count',
         'access callback' => '_member_resource_access',
         'access arguments' => array('get_member_count'),
       ),
     ),
+    'targeted_actions' => array(
+      'password_reset_url' => array(
+        'help' => 'Gets the user password reset URL.',
+        'file' => array(
+          'type' => 'inc', 
+          'module' => 'dosomething_api', 
+          'name' => 'resources/member_resource',
+        ),
+        'args' => array(
+          array(
+            'name' => 'uid',
+            'optional' => FALSE,
+            'source' => array('path' => 0),
+            'type' => 'int',
+            'description' => 'The uid of the user whose password reset url we are getting',
+          ),
+        ),        
+        'callback' => '_member_resource_reset',
+        'access callback' => '_member_resource_access',
+        'access arguments' => array('password_reset_url'),
```

Comment:
Oh, I see, by default it checks for administrator access.  I think this would be a little cleaner if we actually didn't pass an argument to the access callback, and made $op = NULL by default... what do you think @deadlybutter ?